### ISSUE TITLE
Is there a way to get parse() to output if a string is either an expression or a result?

### ISSUE THREAD

**Issue by mofosyne** [josdejong/mathjs#3084]
I've been trying to write a freeform calculator in https://github.com/mofosyne/QuickMathJS and how it currently works is it has a heuristic based function that check if a string portion is either:
 - single variable (to detect straight variable assignment)
 - single function (to detect function assignments)
 - expression ('2x + 2y')
 - result (e.g. '2 km')

Most of my function above is using `math.parse(normalisedStr)` and a series of checks to verify if it is what it is. However verifying results is proving to be a bit tricky, especially if it has matricies or non simple units. I'm just wondering if `math.parse()` already has any straightforward flags just in case I missed it that can do to job already.

This is where isOutputResult() is and how I'm currently approaching it.

https://github.com/mofosyne/QuickMathJS/blob/b712ae5233603c32d98c06a8484128fb6a4b6848/calculator.js#L176-L229

**This is just an inquiry, feel free to close if it doesn't exist and doesn't make sense to include. I already have a working strategy for now, but it would be good to have an official approach**

**Comment by josdejong** [comment]
You can use `math.parse(expression)` and then analyze the returned expression tree to see whether the expression contains a variable, function, or something else.

In your calculator you have a text area that contains a mix of the original expressions, and inserted results. You will need to come up with a solution yourself on how you can recognize what is an expression and what is a result, this is a "problem" that you introduce yourself. I would not put the results on the same line as the expression itself after the `=` (causing `=` to sometimes mean the separator between expression and result, and sometimes be the assignment operator). Like I explained to you via email, You need a simple, reliable way to distinguish expressions and results. For example by putting results on a next line, having an indentation of say 4 spaces. Or by using a different separator between expression and result that does not conflict with the existing operators.

**Comment by mofosyne** [comment]
Thank you for your insights. Yes I recognize that this challenge is certainly self-imposed. However, my goal is to achieve a level of 'naturalness' in user interactions, even if it diverges from the streamlined efficiency that math.js should naturally aim for.

Your suggestion to delve deeper into `math.parse(expression)` has been invaluable. By focusing on analyzing the returned expression tree, I've been able to refine [`isOutputResult()`](https://github.com/mofosyne/QuickMathJS/blob/main/calculator.js#L173-L301), which has worked out so far!

Once again, thanks for the guidance, hopefully if I don't spot any other further issues, then this project should be in a workable state now... and ready for me to see if my approach actually would work out in practice in the real world.

**This issue ticket is now solved**
